### PR TITLE
zh-CN doc translation

### PR DESCRIPTION
**Pull request by cuileon**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes/no
| New feature?  | yes/no
| Breaks BC?    | yes/no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: docs/guide-zh-CN/caching-fragment.md [hunk 1]
```diff
@@ -36,14 +47,17 @@ if ($this->beginCache($id, ['duration' => 3600])) {
 }
 ```
 
-如果该选项未设置，则默认为 0，永不过期。
+If the option is not set, it will take the default value 60, which means the cached content will expire in 60 seconds.
```
Comment (samdark) [inline docs/guide-zh-CN/caching-fragment.md]: Should be translated.

File: docs/guide-zh-CN/caching-data.md [hunk 2]
```diff
@@ -181,44 +262,170 @@ $data = $cache->get($key);
...
+ - `yii cache/flush cache1 cache2`: 冲刷缓存组件`cache1`, `cache2` (可以传递多个用空格分开的缓存组件）
+ - `yii cache/flush-all`: 冲刷应用中所有的缓存组件
+
+> 信息：默认情况下，控制台应用使用独立的配置文件。所以，为了上述命令发挥作用，请确保Web应用和控制台应用配置相同的缓存组件。
 
 
 ### 配置 <span id="query-caching-configs"></span>
 
-查询缓存有两个通过 [[yii\db\Connection]] 设置的配置项：
+Query caching has three global configurable options through [[yii\db\Connection]]:
```
Comment (samdark) [inline docs/guide-zh-CN/caching-data.md]: Should be translated.

File: docs/guide-zh-CN/concept-autoloading.md [hunk 3]
```diff
@@ -19,49 +23,72 @@ Yii 依靠[类自动加载机制](http://www.php.net/manual/en/language.oop5.aut
...
+且类似于 `app\components\MyClass` 这样的类名，
+基于我们刚才所提到的算法，可以正确解析出 `AppBasePath/components/MyClass.php` 路径。
 
-在[高级应用模版](tutorial-advanced-app.md)里，每一逻辑层级会使用他自己的根别名。比如，前端层会使用 `@frontend` 而后端层会使用 `@backend`。因此，你可以把前端的类放在 `frontend` 命名空间，而后端的类放在 `backend`。 这样这些类就可以被 Yii 自动加载了。
+在[高级应用模版](tutorial-advanced-app.md)里，每一逻辑层级会使用他自己的根别名。
+比如，前端层会使用 `@frontend` 而后端层会使用 `@backend`。
+因此，你可以把前端的类放在 `frontend` 命名空间，而后端的类放在 `backend`。 
+这样这些类就可以被 Yii 自动加载了。
 
+To add a custom namespace to the autoloader you need to define an alias for the base directory of the namespace using [[Yii::setAlias()]].
```
Comment (samdark) [inline docs/guide-zh-CN/concept-autoloading.md]: Should be translated.

File: docs/guide-zh-CN/concept-behaviors.md [hunk 4]
```diff
@@ -37,15 +42,22 @@ class MyBehavior extends Behavior
...
+以上代码定义了行为类 `app\components\MyBehavior` 并为要附加行为的组件提供了两个属性 `prop1` 、 `prop2` 和一个方法 `foo()` 。
+注意属性 `prop2` 是通过 getter `getProp2()` 和 setter `setProp2()` 定义的。
+能这样用是因为 [[yii\base\Object]] 是 [[yii\base\Behavior]] 的祖先类，此祖先类支持用 getter 和 setter 方法定义[属性](basic-properties.md)
 
-> Tip: 在行为内部可以通过 [[yii\base\Behavior::owner]] 属性访问行为已附加的组件。
+因为这是一个行为类，当它附加到一个组件时，该组件也将具有 `prop1` 和 `prop2` 属性和 `foo()` 方法。
 
+> 提示：在行为内部可以通过 [[yii\base\Behavior::owner]] 属性访问行为已附加的组件。
+
+> 注意：In case [[yii\base\Behavior::__get()]] and/or [[yii\base\Behavior::__set()]] method of behavior is overridden you
```
Comment (samdark) [inline docs/guide-zh-CN/concept-behaviors.md]: Should be translated.

File: docs/guide-zh-CN/concept-behaviors.md [hunk 5]
```diff
@@ -261,11 +287,14 @@ class User extends ActiveRecord
 
 以上指定的行为数组：
 
-* 当记录插入时，行为将当前时间戳赋值给 `created_at` 和 `updated_at` 属性；
+* 当记录插入时，行为将当前时间戳赋值给 
+  `created_at` 和 `updated_at` 属性；
 * 当记录更新时，行为将当前时间戳赋值给 `updated_at` 属性。
 
+> 注意：For the above implementation to work with MySQL database, please declare the columns(`created_at`, `updated_at`) as int(11) for being UNIX timestamp.
```
Comment (samdark) [inline docs/guide-zh-CN/concept-behaviors.md]: Should be translated.

File: docs/guide-zh-CN/concept-behaviors.md [hunk 6]
```diff
@@ -274,36 +303,58 @@ $user->save();
...
 ```
 
+其它行为
+---------------
+
+There are several built-in and external behaviors available:
 
-与 PHP traits 的比较
--------------------
+- [[yii\behaviors\BlameableBehavior]] - automatically fills the specified attributes with the current user ID.
```
Comment (samdark) [inline docs/guide-zh-CN/concept-behaviors.md]: Should be translated.

File: docs/guide-zh-CN/concept-behaviors.md [hunk 7]
```diff
@@ -274,36 +303,58 @@ $user->save();
...
+- [yii2tech\ar\softdelete\SoftDeleteBehavior](https://github.com/yii2tech/ar-softdelete) - provides methods to soft-delete
+  and soft-restore ActiveRecord i.e. set flag or status which marks record as deleted.
+- [yii2tech\ar\position\PositionBehavior](https://github.com/yii2tech/ar-position) - allows managing records order in an
+  integer field by providing reordering methods.
 
-尽管行为在 "注入" 属性和方法到主类方面类似于 [traits](http://www.php.net/traits) ，它们在很多方面却不相同。如上所述，它们各有利弊。它们更像是互补的而不是相互替代。
+比较行为与 Traits <span id="comparison-with-traits"></span>
+----------------------
 
+While behaviors are similar to [traits](http://www.php.net/traits) in that they both "inject" their
```
Comment (samdark) [inline docs/guide-zh-CN/concept-behaviors.md]: Should be translated.

File: docs/guide-zh-CN/concept-behaviors.md [hunk 8]
```diff
@@ -274,36 +303,58 @@ $user->save();
...
+行为无须修改组件类就可动态附加到组件或移除。
+要使用 traits，必须修改使用它的类。
 
-行为以响应事件来自定义组件的代码执行。
+Behaviors are configurable while traits are not.
 
-当不同行为附加到同一组件产生命名冲突时，这个冲突通过先附加行为的优先权自动解决。而由不同 traits 引发的命名冲突需要通过手工重命名冲突属性或方法来解决。
+Behaviors can customize the code execution of a component by responding to its events.
 
+When there can be name conflicts among different behaviors attached to the same component, the conflicts are
```
Comment (samdark) [inline docs/guide-zh-CN/concept-behaviors.md]: Should be translated.

File: docs/guide-zh-CN/concept-di-container.md [hunk 9]
```diff
@@ -33,9 +40,42 @@ $foo = new Foo($bar);
 ```
 
 
+### Method Injection <span id="method-injection"></span>
+
+Usually the dependencies of a class are passed to the constructor and are available inside of the class during the whole lifecycle.
```
Comment (samdark) [inline docs/guide-zh-CN/concept-di-container.md]: Should be translated.

**Comment by samdark** [comment]
There are many parts that are in English. Is that merge issue or should it be like that? If that's OK, I'll merge as is.

File: docs/guide-zh-CN/blocktypes.json [hunk 10]
```diff
@@ -1,6 +0,0 @@
-{
-    "Warning:": "警告：",
-    "Note:": "注意：",
-    "Info:": "补充：",
-    "Tip:": "提示："
-}
```
Comment (cebe) [inline docs/guide-zh-CN/blocktypes.json]: why is this file deleted?

File: docs/guide-zh-CN/concept-aliases.md [hunk 11]
```diff
@@ -17,17 +18,23 @@ Yii::setAlias('@foo', '/path/to/foo');
 Yii::setAlias('@bar', 'http://www.example.com');
 ```
 
-> Note: 别名所指向的文件路径或 URL 不一定是真实存在的文件或资源。
+> 注意：别名所指向的文件路径或 URL 不一定是真实存在的文件或资源。
```
Comment (cebe) [inline docs/guide-zh-CN/concept-aliases.md]: this change should not be made. keeping "Note:" here will allow formatting the block correctly when docs are rendered. `Note:` will be replaced with `注意：` using the `blocktypes.json` file.

Comment (cebe) [reply docs/guide-zh-CN/concept-aliases.md]: See https://github.com/yiisoft/yii2/blob/master/docs/documentation_style_guide.md#blocks

Comment (cuileon) [reply docs/guide-zh-CN/concept-aliases.md]: OK

File: docs/guide-zh-CN/blocktypes.json (hunk 10 above)
Comment (cuileon) [reply docs/guide-zh-CN/blocktypes.json]: Sorry, that's my mistake, #16061 can be merged。
